lib/portlayer/exec/commit.go

```diff
@@ -196,6 +196,7 @@ func Commit(ctx context.Context, sess *session.Session, h *Handle, waitTime *int
 
 		// start the container
 		if err := c.start(ctx); err != nil {
+			publishContainerEvent(h.ExecConfig.ID, time.Now().UTC(), events.ContainerFailed)
```

Comment:
What is this for?

Reply:
To workaround a VC/ESXi problem. We observed that PowerOn_Task gets stuck on VC time to time even though the VM was starting fine on the host ESXi. Eventually the task was getting timed out (After 20 min.) and that was setting the container state back to Stopped. During that time VC is not generating any other event so the persona listener was getting nothing. This new event is for signaling the eventmonitor so that it can autoremove the container after this failure if it is marked as auto-removable.

Reply:
Might be worth at least a short comment about why the event is being generated - not necessarily the VC part, but noting that this allows async logic hung off the event stream to process the failure.